lib/model.js

```diff
@@ -153,7 +162,16 @@ const _create = (data, definition) => {
       obj._etag = newEtag;
       if (explicitId) { obj._id = explicitId; }
       debug(`inserting into ${collection}: ${JSON.stringify(obj)}`);
-      const result = await db.insert(collection, obj);
+      try {
+        result = await db.insert(collection, obj);
+      } catch (err) {
+        if (err.message.startsWith('E11000 duplicate key') && definition.transformDuplicateKeyError) {
+          // this should throw an error.
+          definition.transformDuplicateKeyError(obj, err);
```

Comment:
I thought about parsing the error message for parts like collection, index and key. 

https://rollbar.com/Losant/losant-workers/items/758/?item_page=0&item_count=100&#traceback